File: src/main/java/work/xeltica/craft/core/models/CraftRecipe.java
package work.xeltica.craft.core.models;

import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

import java.util.ArrayList;
import java.util.Map;
import java.util.stream.Collectors;

public record CraftRecipe(ArrayList<ItemStack> ingredients, ItemStack result) {
    public Map<Material, Integer> getFixedRecipe() {
        return ingredients.stream().collect(
                Collectors.groupingBy(ItemStack::getType, Collectors.summingInt(ItemStack::getAmount))
        );
    }
}

File: src/main/java/work/xeltica/craft/core/models/CraftRecipe.kt
package work.xeltica.craft.core.models

import org.bukkit.Material
import org.bukkit.inventory.ItemStack
import work.xeltica.craft.core.utils.CollectionHelper.sum

@JvmRecord
data class CraftRecipe(val ingredients: ArrayList<ItemStack>, val result: ItemStack) {
    val fixedRecipe: Map<Material, Int>
        get() = ingredients.groupingBy { it.type }.sum { it.amount }
}